pkgs/development/python-modules/pytest-smtpd/default.nix

```diff
@@ -0,0 +1,44 @@
+{
+  lib,
+  buildPythonPackage,
+  pythonOlder,
+  fetchFromGitHub,
+  hatchling,
+  pytest,
+  smtpdfix,
+  pytestCheckHook,
+}:
+
+buildPythonPackage rec {
+  pname = "pytest-smtpd";
+  version = "0.1.0";
+  pyproject = true;
+
+  disabled = pythonOlder "3.8";
+
+  # Pypi tarball doesn't include tests/
+  src = fetchFromGitHub {
+    owner = "bebleo";
+    repo = "pytest-smtpd";
+    rev = "v${version}";
+    hash = "sha256-Vu2D2hfxBYxgXQ4Gjr+jFpac9fjpLL2FftBhnqrcQaA=";
+  };
+
+  build-system = [ hatchling ];
+
+  dependencies = [
+    pytest
```

Comment:
We usually put that in `buildInputs` just in case a package uses a different version of pytest.

Reply:
Mmh interesting. How do I determine that in general? Frequently used packages in `buildInputs`, everything else in `dependencies`?